gnmi/gnmit/datastore.go

```diff
@@ -25,8 +27,26 @@ func NewDatastoreServer(gnmiServer *GNMIServer) *DatastoreServer {
 }
 
 func (d *DatastoreServer) Set(_ context.Context, req *gpb.SetRequest) (*gpb.SetResponse, error) {
-	if err := d.gnmiServer.set(req); err != nil {
+	if err := d.gnmiServer.set(req, false); err != nil {
```

Comment:
should update cache if the setrequest contains config?

Reply:
Thanks for pointing this out, I see two problems here:
1. the datastore should only process state updates, whereas the main gNMI server should only process config updates. Added TODO to enforce this rule since the solution should be similar.
2. We should still verify the validity of the received state values, but currently we ignore state values when unmarshalling due to `PreferShadowPath` so we aren't checking some of the state values being received. I will add another `set()` routine to process these state updates.

Reply:
sgtm with TODO

Reply:
thanks.